pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -126,31 +141,60 @@ function readProfileFiles(rootPath: string, defaultsPath: string, lockedPath: st
  * Windows only. Read settings values from registry using schemaObj as a template.
  * @param schemaObj the object used as a template for navigating registry.
  * @param regKey the registry key obtained from nativeReg.openKey().
- * @param regPath the path to the object relative to regKey.
- * @returns undefined, or the registry data as an object.
+ * @returns null, or the registry data as an object.
  */
-function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY, regPath: string[]): any {
-  let regValue;
-  let newObject: any;
+function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): RecursivePartial<settings.Settings>|null {
+  let newObject: RecursivePartial<settings.Settings>|null = null;
 
-  for (const [schemaKey, schemaVal] of Object.entries(schemaObj)) {
-    if (typeof schemaVal === 'object' && !Array.isArray(schemaVal)) {
-      regValue = readRegistryUsingSchema(schemaVal, regKey, regPath.concat(schemaKey));
-    } else {
-      regValue = nativeReg.getValue(regKey, regPath.join('\\'), schemaKey);
-    }
+  const schemaKeys = Object.keys(schemaObj);
+  // ignore case
+  const registryKeys = nativeReg.enumKeyNames(regKey).concat(nativeReg.enumValueNames(regKey)).map(k => k.toLowerCase());
+  const commonKeys = schemaKeys.filter(k => registryKeys.includes(k.toLowerCase()));
 
-    if (typeof regValue !== 'undefined' && regValue !== null) {
-      newObject ??= {};
+  for (const k of commonKeys) {
+    const schemaVal = schemaObj[k];
+    let regValue: any = null;
+
+    if (typeof schemaVal === 'object') {
+      if (!Array.isArray(schemaVal)) {
+        const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
+
+        if (!innerKey) {
+          continue;
+        }
+        try {
+          regValue = readRegistryUsingSchema(schemaVal, innerKey);
+          if (regValue && (typeof regValue === 'object') && Object.keys(regValue).length === 0) {
```

Comment:
I don't understand how `regValue` can be not `'object'`. Inside the call, there is only one `return` from the end of the function, and `newObject` is always either `null` or `Record<string, any>`.

Reply:
Moved to issue #4153